Source/WebCore/Modules/url-pattern/URLPatternCanonical.cpp

```diff
@@ -216,9 +216,9 @@ ExceptionOr<String> processPathname(StringView pathnameValue, const StringView p
         return pathnameValue.toString();
 
     if (WTF::URLParser::isSpecialScheme(protocolValue))
-        return canonicalizePathname(pathnameValue);
+        return canonicalizePathname(pathnameValue.toString());
```

Comment:
If most call sites on canonicalizePathname() have to call `toString()` then likely `canonicalizePathname()` should keep taking in a `StringView`, rather than a `String&&`

Reply:
It needs to be String&& because otherwise, StringView may somehow release reference.

Reply:
Sounds like you have a bug somewhere then.